src/commands/logStream/logStreamRequest.ts

```diff
@@ -0,0 +1,137 @@
+
+                const genericClient: ServiceClient = await createGenericClient(context, undefined);
+                const headers = createHttpHeaders({
+                    authorization: `Bearer ${token.token}`
+                })
+
+                const logsResponse = await genericClient.sendRequest(createPipelineRequest({
+                    method: "GET",
+                    url: endpoint,
+                    abortSignal: abortController.signal,
+                    headers,
+                    streamResponseStatusCodes: new Set<number>([200])
+                }));
+
+                await new Promise<void>((onLogStreamEnded: () => void, reject: (err: Error) => void): void => {
+                    const newLogStream: ILogStream = {
+                        dispose: (): void => {
+                            logsResponse.readableStreamBody?.removeAllListeners();
+                            abortController.abort();
+                            outputChannel.show();
+                            outputChannel.appendLine(localize('logStreamDisconnected', 'Disconnected from log-streaming service.'));
+                            newLogStream.isConnected = false;
+                            void onLogStreamEnded();
+                        },
+                        isConnected: true,
+                        outputChannel: outputChannel,
+                        data: {
+                            revision: context.revision?.name,
+                            replica: context.replica?.name,
+                            container: context.container?.name,
+                            containerApp: context.containerApp.name,
+                        }
+                    };
+
+                    logsResponse.readableStreamBody?.on('data', (chunk: Buffer | string) => {
+                        outputChannel.append(chunk.toString());
+                    }).on('error', (err: Error) => {
+                        newLogStream.isConnected = false;
+                        outputChannel.show();
+                        outputChannel.appendLine(localize('logStreamError', 'Error connecting to log-streaming service:'));
+                        outputChannel.appendLine(parseError(err).message);
+                        reject(err);
+                    }).on('complete', () => {
+                        newLogStream.dispose();
+                    });
+                    logStreams.set(logStreamId, newLogStream);
+                    onLogStreamCreated(newLogStream);
+                });
+            });
+        });
+    }
+}
+
+export async function disconnectLogStreaming(context: IStreamLogsContext): Promise<void> {
+    const logStream: ILogStream | undefined = context.logStreamToStop;
+    if (!logStream) {
+        const allStreams = getActiveLogStreams(context);
+        for (const streams of allStreams.values()) {
+            if (streams && streams.isConnected) {
```

Comment:
I think the cleaner way to do this is to make this block of code a utility method and then just calling it on logStream or in each stream in `allStreams`.

You could also do something like

`const allStreams = context.logStreamToStop ? [context.logStreamToStop] : getActiveLogStreams(context);`